lib/init/grass.py

```diff
@@ -1160,12 +1160,26 @@ def load_env(grass_env_file):
     if not os.access(grass_env_file, os.R_OK):
         return
 
+    export_re = re.compile('^export[ \t]+([a-zA-Z_]+[a-zA-Z0-9_]*)=(.*?)[ \t]*$')
+
     for line in readfile(grass_env_file).split(os.linesep):
-        try:
-            k, v = map(lambda x: x.strip(), line.strip().split(' ', 1)[1].split('=', 1))
-        except:
+        m = export_re.match(line)
+        if not m:
             continue
-
+        k = m[1]
+        v = m[2]
+        expand = True
```

Comment:
Can you please document the intention here (lines 1171-1182, esp. expand and 1182) by comments in the code? ...so we are not wondering why and what is the code trying to do later on.